src/client/Microsoft.Identity.Client/AuthScheme/PoP/PoPProviderFactory.cs

```diff
@@ -16,22 +16,24 @@ internal class PoPProviderFactory
         private static readonly TimeSpan s_expirationTimespan = TimeSpan.FromHours(8);
         private static object s_lock = new object();
 
-        public static InMemoryCryptoProvider GetOrCreateProvider(/* for testing */ ITimeService timeService = null)
-        {
-            if (timeService == null)
-            {
-                timeService = new TimeService();
-            }
+        internal static ITimeService TimeService { get; set; }
```

Comment:
Small perf improvement - initialize this property with `new TimeService` instead. Tests will simply override it. This will avoid you having to check if it's null at line 25 below.